src/storage/src/storage_state.rs

```diff
@@ -1062,6 +1088,15 @@ impl<'w, A: Allocate> Worker<'w, A> {
             }
         }
 
+        for (id, res) in final_ingestion_description {
+            let existing = &self.storage_state.ingestions[&id];
+            if existing != &res {
+                halt!(
+                    "reconciliation result of ingestion for ID {id} does not match existing ingestion:\nresult: {res:?}\n\nexisting: \n{existing:?}",
```

Comment:
This doesn't look right because reconciliation might happen after the controller sends alter commands and so the rehydration client might send newer state here.

Reply:
Believe I addressed this but please check my work